service/server/main.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import logging
+
+from subprocess import Popen, PIPE, STDOUT, check_call
+
+from flask import Flask, jsonify, request
+
+from headers import VEP_HEADERS
+
+
+# This variant is passed to VEP when `headers` is called. This works
+# even when `--check_ref` is used because the headers line is still produced.
+SAMPLE_VARIANT = 'chr1	807692	.	C	A	0	PASS	KM=11.5	GT	1|1'
+
+
+def setup_vep_cache():
+    """Initialization function to download and build VEP's cache"""
+    # TODO(jessime) These values should be loaded from ENV variables
+    local_cache = '/mnt/vep/vep_cache/'
+    gcs_dir = 'gs://gcp-variant-annotation-vep-cache'
+    vep_file = 'vep_cache_homo_sapiens_GRCh38_91.tar.gz'
+    gcs_vep_file = os.path.join(gcs_dir, vep_file)
+    local_vep_file = os.path.join(local_cache, vep_file)
+    check_call('gsutil cp {} {}'.format(gcs_vep_file, local_cache).split())
+    check_call('tar xzvf {} -C {}'.format(local_vep_file, local_cache).split())
+    # TODO(jessime) Here and elsewhere, replace prints with proper logging
+    print 'VEP CACHE setup.'
+
+
+#Global calls after `setup_vep_cache` is defined but before `app` decorators
```

Comment:
1. If you place `app = Flask(__name__)` after `app` decorators, you'll get a `NameError`, since `app` isn't defined yet. 
2. It should be fine to put the `setup_vep_cache()` behind `if __name__ == '__main__'`. But `app = Flask(__name__)` might cause issues (with Gunicorn, for example, though I would have to investigate more).